artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/federation/AMQPFederationManagementSupport.java

```diff
@@ -141,10 +172,58 @@ public static ObjectName getFederationSourceObjectName(ManagementService managem
             ObjectName.quote(federationName)));
    }
 
+   /**
+    * Register the given {@link AMQPFederationTarget} instance with the broker management services.
+    *
+    * @param federation
+    *    The federation target instance being registered with management.
+    *
+    * @throws Exception if an error occurs while registering the federation with the management services.
+    */
+   public static void registerFederationTarget(String remoteNodeId, String brokerConnectionName, AMQPFederationTarget federation) throws Exception {
+      final String federationName = federation.getName();
+      final ActiveMQServer server = federation.getServer();
+      final ManagementService management = server.getManagementService();
+      final AMQPFederationTargetControlType control = new AMQPFederationTargetControlType(server, federation);
+
+      management.registerInJMX(getFederationTargetObjectName(management, remoteNodeId, brokerConnectionName, federationName), control);
+      management.registerInRegistry(getFederationTargetResourceName(remoteNodeId, brokerConnectionName, federationName), control);
+   }
+
+   /**
+    * Unregister the given {@link AMQPFederationTarget} instance with the broker management services.
+    *
+    * @param federation
+    *    The federation target instance being unregistered from management.
+    *
+    * @throws Exception if an error occurs while unregistering the federation with the management services.
+    */
+   public static void unregisterFederationTarget(String remoteNodeId, String brokerConnectionName, AMQPFederationTarget federation) throws Exception {
+      final String federationName = federation.getName();
+      final ActiveMQServer server = federation.getServer();
+      final ManagementService management = server.getManagementService();
+
+      management.unregisterFromJMX(getFederationTargetObjectName(management, remoteNodeId, brokerConnectionName, federationName));
+      management.unregisterFromRegistry(getFederationTargetResourceName(remoteNodeId, brokerConnectionName, federationName));
+   }
+
+   public static String getFederationTargetResourceName(String remoteNodeId, String brokerConnectionName, String federationName) {
+      return String.format(FEDERATION_TARGET_RESOURCE_TEMPLATE, remoteNodeId, brokerConnectionName, federationName);
+   }
+
+   public static ObjectName getFederationTargetObjectName(ManagementService management, String remoteNodeId, String brokerConnection, String federationName) throws Exception {
+      final String brokerConnectionName = management.getObjectNameBuilder().getRemoteBrokerConnectionBaseObjectNameString(remoteNodeId, brokerConnection);
```

Comment:
Bit of a nitpick, but 'brokerConnectionName' keeps making me think its just the connection name, when its simply not...maybe make it 'brokerConnectionObjectName' for clarity ?

(same in the several other places getRemoteBrokerConnectionBaseObjectNameString is called)